File: java/src/main/java/benchmarks/SortSelf.java
package benchmarks;

import java.util.Arrays;

public class SortSelf extends SortBenchmark {

    @Override
    public String name() {
        return "Sort::Self";
    }

    @Override
    int[] test() {
        int[] arr = data.clone();
        Arrays.sort(arr);
        return arr;
    }
}

File: nim/src/benchmarks/sortself.nim
import std/[algorithm]
import ../benchmark
import ../helper
import sort_common

type
  SortSelf* = ref object of SortBenchmark

proc newSortSelf(): Benchmark =
  SortSelf()

method name(self: SortSelf): string = "Sort::Self"

method test(self: SortSelf): seq[int32] =
  var arr = self.data
  arr.sort(system.cmp[int32])
  arr

registerBenchmark("Sort::Self", newSortSelf)
